torch/nn/modules/_functions.py

```diff
@@ -15,23 +16,18 @@ def forward(self, input, weight, bias, running_mean, running_var, eps, momentum,
         # calculate mean/invstd for input.
         mean, invstd = torch.batch_norm_stats(input, eps)
 
-        count_all = torch.empty(world_size, 1, dtype=count.dtype, device=count.device)
-        mean_all = torch.empty(world_size, mean.size(0), dtype=mean.dtype, device=mean.device)
-        invstd_all = torch.empty(world_size, invstd.size(0), dtype=invstd.dtype, device=invstd.device)
-
-        count_l = list(count_all.unbind(0))
-        mean_l = list(mean_all.unbind(0))
-        invstd_l = list(invstd_all.unbind(0))
-
-        # using all_gather instead of all reduce so we can calculate count/mean/var in one go
-        count_all_reduce = torch.distributed.all_gather(count_l, count, process_group, async_op=True)
-        mean_all_reduce = torch.distributed.all_gather(mean_l, mean, process_group, async_op=True)
-        invstd_all_reduce = torch.distributed.all_gather(invstd_l, invstd, process_group, async_op=True)
-
-        # wait on the async communication to finish
-        count_all_reduce.wait()
-        mean_all_reduce.wait()
-        invstd_all_reduce.wait()
+        num_channels = input.shape[1]
+        # C, C, 1 -> (2C + 1)
+        combined = torch.cat([mean, invstd, count], dim=0)
+        # world_size * (2C + 1)
+        combined_list = [
+            torch.zeros_like(combined) for k in range(world_size)
```

Comment:
you don't need zeros_like here (it fills tensors with zeros) because values will be overwritten, right? torch.empty_like is faster and should be enough.